source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -200,11 +200,14 @@ void RawHttpClientImpl::check(RequestCallbacks& callbacks,
 
   for (const auto& header : request.attributes().request().http().headers()) {
     const Http::LowerCaseString key{header.first};
+    // Skip setting content-length header since it is already configured at initialization.
+    if (key == Http::Headers::get().ContentLength) {
```

Comment:
A few questions:
1. Why is `content-length` alone unique? Are there any other similar headers?
2. Where is this documented in the API or operator docs? I think we need to explain this.
3. What harm is there in sending `content-length` to the ext authz server, even if it is redundant with request size?

Reply:
1. Because the `content-length` of the request to be sent to the authorization server is not the request itself, but it is the "check request". The payload of the original request can be forwarded in the check request, but it depends on the [max_request_bytes](https://github.com/envoyproxy/envoy/blob/c7f25e8f1d94c72d952544e46ae379b928883ea7/api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto#L112) and if it is [allowed to send partial message or not](https://github.com/envoyproxy/envoy/blob/c7f25e8f1d94c72d952544e46ae379b928883ea7/api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto#L117). Hence it is possible the `content-length` is different (smaller) than the original request.
2. I agree that wee need to explain this behavior more in the doc, it is now cluttered in several places, probably here: https://github.com/envoyproxy/envoy/blob/c7f25e8f1d94c72d952544e46ae379b928883ea7/api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto#L168-L169 and here: https://github.com/envoyproxy/envoy/blob/c7f25e8f1d94c72d952544e46ae379b928883ea7/api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto#L56-L59
3. The `content-length` should be matched with the payload size as mentioned in point 1. above. This patch however to avoid the bugs as mention in the #9603, (the check request has `content-length` header contains `,` in its value, e.g. `content-length: 0,14`), since the value of `content-length` of the check request is already initialized in [here](https://github.com/envoyproxy/envoy/blob/20dddf007f05dd5ad8d09f7221fe835fb5699cfd/source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc#L190-L199).

Reply:
Ack, I think if you can just add some docs as described in (2) this sounds good.

Reply:
👍